samcli/lib/config/samconfig.py

```diff
@@ -144,30 +146,29 @@ def put_comment(self, comment):
         comment: str
             A comment to write to the samconfg file
         """
-        if self.document is None:
-            self._read()
 
-        self.document.add(tomlkit.comment(comment))
+        self.document.update({"__comment__": comment})
```

Comment:
Did you verify that the TOML library knows to handle this line as a comment?

Reply:
I assumed the TOML library would, but looking through, I think it may have to be manually commented in through the `TomlFileManager.write()` method. Will update accordingly 👍